### ISSUE TITLE
Create integration tests to verify the extension TLS behavior

### ISSUE THREAD

**Issue by cwperks** [opensearch-project/opensearch-sdk-java#715]
https://github.com/opensearch-project/opensearch-sdk-java/pull/619/files introduced settings that setup extension to run with TLS and connect with an OpenSearch node securely. In the security plugin, there are a suite of SSL Tests (See [SSLTest](https://github.com/opensearch-project/security/blob/main/src/test/java/org/opensearch/security/ssl/SSLTest.java)) that verify the different TLS settings and verify cluster formation using TLS. For extensions, there should be a similar suite of tests with the different settings provided in SSLConfigConstants to verify that an OpenSearch node with the security plugin installed can communicate with an extension over a secure TLS channel.

**Comment by minalsha** [comment]
@ryanbogan  @dbwiddis , @saratvemulapalli , @owaiskazi19  thoughts?

**Comment by dbwiddis** [comment]
I'm in favor.  @cwperks would love you to help create these tests.

**Comment by owaiskazi19** [comment]
It's always good to have more tests.

**Comment by ryanbogan** [comment]
I agree as well.  We should definitely have tests that ensure security and extensions are working together properly.
